api/v1beta2/awscluster_webhook.go

```diff
@@ -190,3 +191,19 @@ func (r *AWSCluster) validateNetwork() field.ErrorList {
 	}
 	return allErrs
 }
+
+func (r *AWSCluster) validateAdditionalIngressRules() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if r.Spec.ControlPlaneLoadBalancer == nil {
+		return allErrs
+	}
+
+	for _, rule := range r.Spec.ControlPlaneLoadBalancer.AdditionalIngressRules {
+		if (rule.CidrBlocks != nil || rule.IPv6CidrBlocks != nil) && (rule.SourceSecurityGroupIDs != nil || rule.SourceSecurityGroupRoles != nil) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("additionalIngressRules"), r.Spec.ControlPlaneLoadBalancer.AdditionalIngressRules, "CIDR blocks and security group IDs cannot be used together"))
```

Comment:
It is not only `SourceSecurityGroupIDs` as per the condition, so probably should be something like:
```suggestion
			allErrs = append(allErrs, field.Invalid(field.NewPath("additionalIngressRules"), r.Spec.ControlPlaneLoadBalancer.AdditionalIngressRules, "CIDR blocks and security group IDs or security group roles cannot be used together"))
```